lib/grape/dsl/inside_route.rb

```diff
@@ -94,7 +94,13 @@ def redirect(url, options = {})
       #
       # @param status [Integer] The HTTP Status Code to return for this request.
       def status(status = nil)
-        if status
+        if status.is_a? Symbol
+          if Rack::Utils::SYMBOL_TO_STATUS_CODE.keys.include?(status)
+            @status = Rack::Utils.status_code(status)
+          else
+            fail ArgumentError, "Status code :#{status} is not valid."
```

Comment:
You want to specialize this error and make it localizable too, https://github.com/intridea/grape/blob/master/lib/grape/locale/en.yml.